src/Symfony/Component/Form/Tests/Extension/Core/Type/DateTypeTest.php

```diff
@@ -1156,6 +1156,72 @@ public function testDateTimeImmutableInputTimezoneNotMatchingModelTimezone()
         ]);
     }
 
+    public function testSubmitWithCustomCalendarOption()
+    {
+        $calendar = \IntlCalendar::createInstance();
+        $calendar->setFirstDayOfWeek(\IntlCalendar::DOW_SUNDAY);
+
+        $form = $this->factory->create(static::TESTED_TYPE, options: [
+            'format' => "y-MM-dd y'w'w",
+            'html5' => false,
+            'input' => 'array',
+            'calendar' => $calendar,
+        ]);
+        $form->submit('2024-03-31 2024w14');
```

Comment:
Should we add failing tests when submitting `2024-03-31 2024w13` with this calendar and `2024-03-31 2024w14` with the calendar configured on line 1179?

Reply:
I'm not sure, which case would you like to test exactly by doing this?

Reply:
I am not sure what would happen then. Would the form accept such an invalid behaviour? In the best case the transformer would issue a `TransformationFailedException` and we could test for the form being invalid.

Reply:
I had a bit of trouble with the weeks. I reworked the tests instead to use another calendar instead of playing on the edge of weeks (which also didn't seem to be really supported by this formtype). What do you think of the new version?